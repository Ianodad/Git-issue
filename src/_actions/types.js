

// auth actions 
export const SIGN_UP = "SIGN_UP"
export const SIGN_IN = "SIGN_IN"

 // api actions
export const GET_SEARCH_USER = "GET_SEARCH_USER"
export const GET_ALL_OWNER_REPOS = "GET_ALL_USER_REPOS"
export const GET_ALL_OWNER_ISSUES = "GET_ALL_OWNER_ISSUES"
export const GET_ALL_OWNER_REPO_ISSUES = "GET_ALL_OWNER_REPO_ISSUES"
export const GET_ONE_OWNER_ISSUE = "GET_ONE_OWNER_ISSUE"
export const GET_REPO_ISSUES_COMMENTS = "GET_REPO_ISSUES_COMMENTS"
export const GET_REPO_ISSUE_COMMENTS = "GET_REPO_ISSUE_COMMENTS"

// user actions
export const SELECT_USER = "SELECT_USER"