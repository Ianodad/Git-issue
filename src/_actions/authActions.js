import firebase from "../config/firebase";

import {
  SIGN_IN_ANONYMOUSLY,
  SIGN_IN_WITH_GITHUB,
  SIGN_OUT,
  GET_USER_STATE,
} from "./types";

const SignInWithGithub = (history) => async (dispatch) => {
  
  console.log("github");
  const provider = new firebase.auth.GithubAuthProvider();

  // await firebase.auth().signInWithRedirect(provider);
  try {
    const { additionalUserInfo } = await firebase
      .auth()
      .signInWithPopup(provider);
    const user = additionalUserInfo.profile;

    //  console.log(user.login)
    const userGithub = {
      login: user.login,
      avatar_url: user.avatar_url,
      html_url: user.html_url,
      id: user.id,
      name: user.name,
      repos_url: user.repos_url,
      type: user.type,
    };

    if (user) {
      localStorage.setItem('USER', JSON.stringify(userGithub));
      dispatch({
        type: SIGN_IN_WITH_GITHUB,
        payload: userGithub,
      });

      history.push('/')
      // window.location = `/`;
    }

    // if (userGithub) {
    // }
  } catch (error) {
    console.log(error);

    dispatch({
      type: SIGN_IN_WITH_GITHUB,
      payload: null,
    });
  }

  //  console.log(additionalUserInfo)
  dispatch({
    type: SIGN_IN_WITH_GITHUB,
    payload: "sign up",
  });
};

const SignInAnonymous = () => async (dispatch) => {
  await firebase.auth().signInAnonymously();

  // window.location=`/`
  dispatch({
    type: SIGN_IN_ANONYMOUSLY,
    payload: "sign in",
  });
};

const SignOut = () => async (dispatch) => {
  // console.log("sign")
  try {
    await firebase.auth().signOut();
    localStorage.removeItem("USER");

    dispatch({
      type: SIGN_OUT,
      payload: {},
    });

    window.location = `/auth`;
  } catch (error) {
    console.log(error);
  }
};


const getUserState = () => async (dispatch) => {
  const userState = await JSON.parse(localStorage.getItem("USER"));
  // console.log(await JSON.parse(localStorage.getItem("USER")))
  console.log(userState)

  if (userState) {
    dispatch({
      type: GET_USER_STATE,
      payload: userState,
    });
  }
};
// export all action functions as an object
export const authActions = {
  SignInWithGithub,
  SignInAnonymous,
  getUserState,
  SignOut,
};
