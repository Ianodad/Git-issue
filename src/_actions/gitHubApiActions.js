
import gitHubApi from "../api/gitHubApi";
import {GET_SEARCH_USER, GET_ALL_OWNER_REPOS,  GET_ALL_OWNER_ISSUES, GET_ALL_OWNER_REPO_ISSUES, GET_ONE_OWNER_ISSUE, GET_REPO_ISSUES_COMMENTS ,GET_REPO_ISSUE_COMMENTS} from "./types";



// NOTE settings
// default username == octocat
// default repo == "Hello-World"
// search user 
const getSearchUser = (query) => async (dispatch) => {

  try {
    const {data} = await gitHubApi.getSearchUser(query)
    localStorage.setItem("userSearchResult", JSON.stringify(data.items))

    dispatch({
       type: GET_SEARCH_USER,
       payload: data.items,
     });

   if (data) {
    window.location=`/search/${query}`
  }
  } catch (error) {
    console.log(error) 
  }
   

}
// this function is responsible for get all user repos.. base on github api
const getAllOwnerRepos =(owner) => async (dispatch) => {
  const {data} = await gitHubApi.getAllOwnerRepos(owner)
  // console.log(data)
  dispatch({
      type: GET_ALL_OWNER_REPOS ,
      payload: data,
    });
}
// get all repo issues.
const  getAllOwnerRepoIssues = (owner, repo) => async (dispatch) => {
    const {data}= await gitHubApi.getAllOwnerRepoIssues(owner, repo)

    dispatch({
      type: GET_ALL_OWNER_REPO_ISSUES,
      payload: data,
    });
};


// get specific issue straight from redux state..
const getFromDispatch = (number) => async (dispatch, getState) => {
  // getAllOwnerRepoIssues()
  const data = await getState().gitHubApiData.allOwnerRepoIssues;

  try {
    const result = data.find(item=>item.number == number)
    dispatch({
        type:GET_ONE_OWNER_ISSUE,
        payload: result,
      });
    
  } catch (error) {
    console.log(error)  
  }
  // console.log(data);
  // console.log(result)

  
}

export const gitHubApiActions = {
    getAllOwnerRepoIssues,
    getAllOwnerRepos,
    getFromDispatch,
    getSearchUser
};