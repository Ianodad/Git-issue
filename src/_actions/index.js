export * from './gitHubApiActions'
export * from './authActions'
