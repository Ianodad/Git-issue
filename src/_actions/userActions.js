import {SELECT_USER} from "./types";
export * from './gitHubApiActions'



const selectUser = (user) => {
    
}
