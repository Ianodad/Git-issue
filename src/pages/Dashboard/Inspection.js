/* eslint-disable jsx-a11y/img-redundant-alt */
import React, { Component } from "react";
import Layout from "../../Layouts/Layout";
import { Link } from 'react-router-dom';

import { gitHubApiActions } from "../../_actions";

import { connect } from "react-redux";

const { getAllOwnerRepoIssues, getFromDispatch, getRepoIssueComments }= gitHubApiActions;

class Inspection extends Component {
   componentDidMount= async()=>{
        
        const owner = this.props.match.params.owner
        const repo = this.props.match.params.repo
        const id = this.props.match.params.id
        //  this.props.getAllOwnerRepoIssues()
         this.props.getFromDispatch(id)
         this.props.getRepoIssueComments(owner, repo, id)
      // this.setState({data: await this.props.getAllOwnerRepoIssues() })
    }


  render() {
    const {issue} = this.props
    console.log(issue)
    if (!issue){
      return (
        <Layout>
          <div className="container-fluid text-center" style={{marginTop:"25%"}}>Nothing to show please select and issue form <span><Link to={"/issues"}>Issues</Link></span></div>
        </Layout>
      )
    } 
    return (
      <>
        <Layout>
          <div className="container-fluid">
            <div className="card mt-4">
              <div className="card-header" style={{backgroundColor: '#F1F0FF'}}>
                <div className="header row">
                <div className="inspection-title col">
                  <h2 style={{fontWeight:"900", color:"#443266"}}>{issue.title}</h2>
                </div>
                <div className="actions row float-right mx-3">
                  <div className="col">link</div>
                  <div className="col">github</div>
                  <div className="col">edit</div>
                </div>
                </div>
              </div>
              <div className="card-body">
                <div className="info row d-flex">
                  <div className="col">
                    <h3 style={{fontWeight:"900", color:"#443266"}}>Date</h3>
                    <p>{issue.created_at}</p>
                  </div>
                  <div className="col">
                    <h4 style={{fontWeight:"900", color:"#443266"}}>State</h4>
                    <p>{issue.state}</p>
                  </div>
                  <div className="col">
                    <h3 style={{fontWeight:"900", color:"#443266"}}>Locked</h3>
                    <p>{issue.locked}</p>
                  </div>
                </div>
                <div className="disc">
                  <h3 style={{fontWeight:"900", color:"#443266"}}>Description</h3>
                  <p>
                    {issue.body}
                  </p>
                </div>
                <div className="repo-detail">
                  <h3 style={{fontWeight:"900", color:"#443266"}}>Repository Details</h3>
                </div>
              </div>
            </div>
              {/* <div className="title ml-4 my-2">
                <h2 style={{fontWeight:"900", color:"#443266"}}>Comments</h2>
              </div>
            <div className="card mt-2 col-md-10" style={{ width: "75%" }}>
              <div className="card-body">
              <div className="commments">
                <div className="row" >
                  <div className="col-1">
                 <img
                      className="img-card"
                      src="https://via.placeholder.com/75C/O https://placeholder.com/"
                      alt="profile image"
                    ></img>
                  </div>
                  <div className=" col-11">
                    <h5 className="card-title">Card title</h5>
                    <p className="card-text">
                      Some quick example text to build on the card title and
                      make up the bulk of the card's content.
                    </p>
                  </div>
                </div>
              </div>
              </div>
            </div> */}
          </div>
        </Layout>
      </>
    );
  }
}

const mapStateToProps = (state) => {
  console.log(state.gitHubApiData)
  return {
    issue: state.gitHubApiData.oneOwnerIssue,
  };
};

export default connect(mapStateToProps, {
  getAllOwnerRepoIssues,
  getRepoIssueComments,
  getFromDispatch
})(Inspection);
