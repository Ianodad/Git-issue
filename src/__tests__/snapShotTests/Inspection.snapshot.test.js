import React from "react";
import { BrowserRouter, Switch } from 'react-router-dom';

import { render, fireEvent, cleanup } from "@testing-library/react";
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

import Inspection from "../../pages/Dashboard/Inspection";

import thunk from 'redux-thunk';
import {combineReducers} from 'redux';
import rootReducer from '../../_reducers';

import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { gitHubApiActions } from "../../_actions";
import {GET_SEARCH_USER, GET_ALL_OWNER_REPOS, GET_ALL_OWNER_ISSUES, GET_ALL_OWNER_REPO_ISSUES, GET_ONE_OWNER_ISSUE, GET_REPO_ISSUES_COMMENTS ,GET_REPO_ISSUE_COMMENTS } from "../../_actions/types";

const { getFromDispatch, getRepoIssueComments }= gitHubApiActions;

const middlewares = [thunk];
const mockStore = configureStore(middlewares);


describe("<Inspection/> testing component", () => {
  let store;
  let component;    
  let state = {
    gitHubApiData:{
      repoIssues:"",
      allOwnerRepos:"",
      allOwnerRepoIssues: "",
      // oneOwnerIssue: "",
      }
    }
    beforeEach(() => {
    store = mockStore(state);

     // Dispatch the action
    // store.dispatch(getAllOwnerRepoIssues())
     store.dispatch = (getFromDispatch());

    component = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Switch>
            <Inspection/>
          </Switch>
        </BrowserRouter>
      </Provider>
    );
  });

    describe("Check component renders", () => {
         it('should render with given state from Redux store', () => {
          expect(component.toJSON()).toMatchSnapshot();
        });
    });
});