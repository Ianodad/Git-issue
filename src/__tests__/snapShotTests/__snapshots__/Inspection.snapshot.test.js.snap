// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Inspection/> testing component Check component renders should render with given state from Redux store 1`] = `
<div
  className="container-fluid"
  id="layout"
>
  <div
    className="col-md-12 row py-0"
    id="top-section"
    style={
      Object {
        "height": "60px",
      }
    }
  >
    <div
      className="col-md-2 col-12 row mx-auto py-2"
      id="layout-logo"
    >
      <div
        className="col-2 my-auto"
      >
        <h1
          className="mx-auto"
          id="logo-main"
          style={
            Object {
              "fontSize": "3vh",
            }
          }
        >
          gitIssue
        </h1>
      </div>
    </div>
    <div
      className="col-md-10 layout"
      id="layout-menu"
    >
      <nav
        className="navbar navbar-expand-md  mt-1"
        style={
          Object {
            "height": "55px",
          }
        }
      >
        <div
          className="collapse navbar-collapse col"
          id="navbarCollapse"
        >
          <form
            className="form-inline"
            role="form"
          >
            <div
              className="form-group"
            >
              <input
                className="form-control"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Search"
                required="required"
                type="text"
              />
            </div>
            <div
              className="input-group-append ml-1"
            >
              <button
                className="btn btn-block"
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "#443266",
                    "color": "white",
                  }
                }
                type="submit"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                  />
                </svg>
              </button>
            </div>
          </form>
        </div>
        <div
          className="navbar-nav"
        >
          <p
            className="nav-item col"
          >
            Username
          </p>
          <a
            className="nav-item col"
            href="#"
            onClick={[Function]}
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M5 22a1 1 0 0 1-1-1V3a1 1 0 0 1 1-1h14a1 1 0 0 1 1 1v3h-2V4H6v16h12v-2h2v3a1 1 0 0 1-1 1H5zm13-6v-3h-7v-2h7V8l5 4-5 4z"
                />
              </g>
            </svg>
          </a>
        </div>
      </nav>
    </div>
  </div>
  <div
    className="col-md-12 row "
    id="bottom-section"
  >
    <div
      className="col-md-2 px-0"
      id="layout-sidemenu"
      style={
        Object {
          "height": 703,
        }
      }
    >
      <div
        className="list-group borderless mt-4 border-0 mx-0"
        id="sidebar"
      >
        <a
          aria-current="page"
          className="list-group-item list-group-item-action border-0 bg-transparent active"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "color": "grey",
            }
          }
        >
          <i
            className="fa fa-home"
          />
           Main
        </a>
        <a
          aria-current={null}
          className="list-group-item list-group-item-action border-0 bg-transparent "
          href="/issues"
          onClick={[Function]}
          style={
            Object {
              "color": "#8C489F",
            }
          }
        >
          <i
            className="fa fa-camera"
          />
           Issues
        </a>
        <a
          aria-current={null}
          className="list-group-item list-group-item-action border-0 bg-transparent"
          href="/inspection"
          onClick={[Function]}
          style={
            Object {
              "color": "#8C489F",
            }
          }
        >
          <i
            className="fa fa-music"
          />
           Inspection
        </a>
      </div>
    </div>
    <div
      className="col-md-10"
      id="layout-dashboard overflow-auto"
      style={
        Object {
          "height": 703,
        }
      }
    >
      <div
        className="container-fluid text-center"
        style={
          Object {
            "marginTop": "25%",
          }
        }
      >
        Nothing to show please select and issue form 
        <span>
          <a
            href="/issues"
            onClick={[Function]}
          >
            Issues
          </a>
        </span>
      </div>
    </div>
  </div>
</div>
`;
